src/QsCompiler/Compiler/CompilationLoader.cs

```diff
@@ -569,12 +569,11 @@ public CompilationLoader(SourceLoader loadSources, ReferenceLoader loadReference
             PerformanceTracking.TaskStart(PerformanceTracking.Task.RewriteSteps);
             var steps = new List<(int, string, Func<QsCompilation?>)>();
             var qirEmissionEnabled = this.externalRewriteSteps.Any(step => step.Name == "QIR Generation");
-
             if (this.config.IsExecutable && !this.config.SkipSyntaxTreeTrimming)
             {
                 // TODO: It would be nicer to trim unused intrinsics. Currently, this is not possible due to how the old setup of the C# runtime works.
                 // With the new setup (interface-based approach for target packages), it is possible to trim ununsed intrinsics.
-                var rewriteStep = new LoadedStep(new SyntaxTreeTrimming(keepAllIntrinsics: !qirEmissionEnabled), typeof(IRewriteStep), thisDllUri);
+                var rewriteStep = new LoadedStep(new SyntaxTreeTrimming(keepAllIntrinsics: true), typeof(IRewriteStep), thisDllUri);

```

Comment:
Are the two booleans passed into `SyntaxTreeTrimming` and `Monomorphization` reversed?

Reply:
Yes, they are. I would align them if one of them is new such that they are not reversed, but if both already exist I don't think it's worth making the breaking change. That's a good thing to watch out for when reviewing the feature branch as a whole.